libs/authorizenet/src/drivers/authorize-net.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable, Inject } from '@angular/core';
+import { Store } from '@ngrx/store';
+
+	import { 
+		DaffPaymentServiceInterface,
+		DaffPaymentTransformer,
+		DaffPaymentTransformerService
+	} from '@daffodil/checkout';
+
+import { DaffAuthorizeNetGenerateTokenSuccess, DaffAuthorizeNetGenerateTokenFailure } from '../actions/authorizenet.actions';
+import { AuthorizeNetResponse } from '../models/response/authorize-net-response';
+import { DaffAuthorizeNetTokenRequest } from '../models/request/authorize-net-token-request';
+
+declare var Accept: any;
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffAuthorizeNetPaymentService implements DaffPaymentServiceInterface<DaffAuthorizeNetTokenRequest> {
+
+	constructor(
+		@Inject(DaffPaymentTransformer) public paymentTransformer: DaffPaymentTransformerService<DaffAuthorizeNetTokenRequest>,
+		private store: Store<any>
+	) {}
+
+	generateToken(paymentTokenRequest: DaffAuthorizeNetTokenRequest): string {
+		Accept.dispatchData(this.paymentTransformer.transformOut(paymentTokenRequest), this.responseHandler.bind(this));
+
+		return 'this string needs to be returned because authorize.net does not have a normal api';
```

Comment:
Can this just be a void function instead of returning a string?

Reply:
I don't think so, because the `DaffPaymentService.generateToken()` _should_ ideally return a string. And I think that for most payment processor APIs, it should be able to just fine. It's just that the authorize.net API is weird and makes that impossible. So to return void, I'd need to change the `DaffPaymentService` to something like
```
generateToken(paymentTokenRequest: DaffPaymentTokenRequest): string | void;
```
And I'd rather contain the weirdness to authorize.net instead